lib/MusicBrainz/Server/Edit/Artist/Merge.pm

```diff
@@ -51,11 +56,61 @@ sub do_merge
 {
     my $self = shift;
 
-    $self->c->model('Artist')->merge(
-        $self->new_entity->{id},
-        [ $self->_old_ids ],
+    my $new_id = $self->new_entity->{id};
+    my @old_ids = $self->_old_ids;
+    my $all_ids = [$new_id, @old_ids];
+
+    my (undef, $dropped_columns) = $self->c->model('Artist')->merge(
+        $new_id,
+        \@old_ids,
         rename => $self->data->{rename}
     );
+
+    if ($dropped_columns->{type}) {
+        my $dropped_type =
+            $self->c->model('ArtistType')->get_by_id($dropped_columns->{type});
+
+        $self->c->model('EditNote')->add_note(
+            $self->id => {
+                editor_id => $EDITOR_MODBOT,
+                text => localized_note(
+                    'The “{artist_type}” type has been dropped because ' .
+                    'one of the artists has a gender.',
```

Comment:
"The “{artist_type}” type has been dropped because one of the artists has a gender." -> The “{artist_type}” type has not been added to the destination artist because it conflicted with the gender setting on one of the artists being merged" maybe? This otherwise suggests that type and gender just can't ever coexist, to me.

Reply:
It suggests that “{artist_type}” type can’t ever coexist with a gender, at least to me.
But I agree with @reosarevok on replacing “been dropped” with “not been added to the destination artist”.

Reply:
> This otherwise suggests that type and gender just can't ever coexist, to me.

Maybe we ought to explicitly state that "Group artists cannot have a gender" at the end. 'Cause I could also see how "... because it conflicted with the gender setting on one of the artists" might imply it works with some genders but not all.

And I wasn't sure about specifying "artists being merged" rather than just "artists" 'cause I usually only think of source artists as being merged. So what about:

> The “{artist_type}” type has not been added to the destination artist because it conflicted with the gender setting on one of the artists here. Group artists cannot have a gender.

Reply:
Seems fine to me :)